app/models/relationship.rb

```diff
@@ -27,7 +27,7 @@ def self.filter_by_resource_type(relationships, options)
     of_type = Array.wrap(options[:of_type])
     except_type = Array.wrap(options[:except_type])
     return relationships if of_type.empty? && except_type.empty?
-    if relationships.kind_of?(Array)
+    if relationships.kind_of?(Array) || relationships.try(:loaded?)
```

Comment:
@chessbyte this is testing 2 different things:

1. if it is an array, then we want to do the filtering in memory
2. If this is an association, but it is in memory - then we want to do the filtering in memory.

otherwise, lets do the filtering in sql.

Does that shed some light onto the existing code? Should I add a comment to make this more clear?

I'm not sure how I feel about `respond_to?(:reject)` - think that may be different than the original intent